src/integrationTest/java/integration/tests/PreparedStatementTest.java

```diff
@@ -555,6 +556,72 @@ void shouldFetchSpecialCharacters() throws SQLException, MalformedURLException {
 		}
 	}
 
+	@Test
+	void shouldFetchBoolean() throws SQLException {
+		try (Connection connection = createConnection()) {
+			try (PreparedStatement statement = connection
+					.prepareStatement("SELECT ? as a, ? as b, ? as c")) {
+				statement.setBoolean(1, true);
+				statement.setObject(2, false);
+				statement.setObject(3, true, Types.BOOLEAN);
+				statement.execute();
+				ResultSet rs = statement.getResultSet();
+				assertEquals(FireboltDataType.BOOLEAN.name().toLowerCase(),
+						rs.getMetaData().getColumnTypeName(1).toLowerCase());
+				assertEquals(FireboltDataType.BOOLEAN.name().toLowerCase(),
+						rs.getMetaData().getColumnTypeName(2).toLowerCase());
+				assertEquals(FireboltDataType.BOOLEAN.name().toLowerCase(),
+						rs.getMetaData().getColumnTypeName(3).toLowerCase());
+				assertTrue(rs.next());
+				assertTrue(rs.getBoolean(1));
+				assertFalse(rs.getBoolean(2));
+				assertTrue(rs.getBoolean(3));
+			}
+		}
+	}
+
+	@Ignore
+	@ParameterizedTest
+	@MethodSource("booleanTypes")
+	void shouldFetchBooleanFromVariousObjects(Object objectTrue, Object objectFalse) throws SQLException {
+		try (Connection connection = createConnection()) {
+			try (PreparedStatement statement = connection
+					.prepareStatement("SELECT ? as a, ? as b")) {
+				statement.setObject(1, objectTrue, Types.BOOLEAN);
+				statement.setObject(2, objectFalse, Types.BOOLEAN);
+				statement.execute();
+				ResultSet rs = statement.getResultSet();
+				assertEquals(FireboltDataType.BOOLEAN.name().toLowerCase(),
+						rs.getMetaData().getColumnTypeName(1).toLowerCase());
+				assertEquals(FireboltDataType.BOOLEAN.name().toLowerCase(),
+						rs.getMetaData().getColumnTypeName(2).toLowerCase());
+				assertTrue(rs.next());
+				assertTrue(rs.getBoolean(1));
+				assertFalse(rs.getBoolean(2));
+			}
+		}
+	}
+
+	Stream<Arguments> booleanTypes() {
```

Comment:
We're duplicating this method. Is there a way to reuse it in different tests? Can we add it to the parent class?